src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -175,7 +175,78 @@ internal class DaprClientGrpc : DaprClient
             }
         }
 
+        /// <inheritdoc/>
+        public override Task<BulkPublishResponse> BulkPublishEventAsync<TValue>(
+            string pubsubName,
+            string topicName,
+            IReadOnlyList<TValue> events,
+            Dictionary<string, string> metadata = default,
+            CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(pubsubName, nameof(pubsubName));
+            ArgumentVerifier.ThrowIfNullOrEmpty(topicName, nameof(topicName));
+            ArgumentVerifier.ThrowIfNull(events, nameof(events));
+            return MakeBulkPublishRequest(pubsubName, topicName, events, metadata, cancellationToken);
+        }
+        
+        private async Task<BulkPublishResponse> MakeBulkPublishRequest<TValue>(
+            string pubsubName,
+            string topicName,
+            IReadOnlyList<TValue> events,
+            Dictionary<string, string> metadata,
```

Comment:
qualify this specifically as `requestMetadata`?

Reply:
We've used metadata elsewhere so I kept the same convention.